Develop2

EryominAlexander opened the pull request:
Созданы api - тесты на методы:
- POST /api/auth/register создание пользователя
- POST /api/auth/logi логин пользователя
- PATCH /api/auth/user редактироание данных пользователя
- POST /api/orders создание заказа
- GET /api/orders получение заказов пользователей
Для ресурса https://stellarburgers.nomoreparties.site/

EryominAlexander committed:
Настройка pom.xml, подключены junit 4, restassured и allure

EryominAlexander committed:
Добавлены тесты на метод создания пользователя POST /api/auth/register

EryominAlexander committed:
Добавлены тесты для метода авторизации пользователей POST /api/auth/login

EryominAlexander committed:
Добавлена библиотека запросов

EryominAlexander committed:
Добавлены тесты для метода редактирование пользователя PATCH /api/auth/user

EryominAlexander committed:
Добавлены тесты для метода создания заказа POST /api/orders

EryominAlexander committed:
Добавлены тесты для метода получения заказов пользователя GET /api/orders

EryominAlexander committed:
add allure report

egorvas commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,164 @@
+        String token = apiCore.getToken(testEmail, testPassword);
+
+        List<String> idList = new ArrayList<>();
+        idList.add("61c0c5a71d1f82001bdaaa6d");
```
**Нужно исправить:** Тестовые стенды часто сбрасываются, меняются и тп. Нельзя полагаться на данные из них в тестах, потому что в какой-то момент твои тесты просто перестанут работать. Хочешь использовать какие-то данные в тесте - создай.

egorvas commented on src/test/java/CreateOrderTest.java:
```diff
@@ -0,0 +1,164 @@
+    public void setUp() {
+        RestAssured.baseURI = BASE_URL;
+        Random random = new Random();
+        testEmail = "testEmail" + random.nextInt(1000000) + "@yandex.ru";
```
**Можно лучше:** Вместо того чтобы самому придумывать тестовые строки я рекомендую использовать модули для генерации данных, например java faker: https://github.com/DiUS/java-faker

EryominAlexander committed:
Исправлена генерация данных пользователей.
Исправлена генерация тестовых данных для запросов создания заказов. Ранее при созданнии запроса брался статический id ингредиента, теперь создана функция которая возвращает список id ингредиентов, взятых из ответа на запрос.

EryominAlexander committed:
Добавлен новый отчёт
